api/client.go

```diff
@@ -406,6 +416,14 @@ func (c *Config) ReadEnvironment() error {
 		envHTTPProxy = v
 	}
 
+	if v := os.Getenv(EnvReturnErrorOnMissing); v != "" {
+		var err error
+		envReturnErrorOnMissing, err = strconv.ParseBool(v)
```

Comment:
Typically with environment variables that are meant to toggle some behavior on or off, any non-empty value is considered true and an empty value is considered false. Parsing the given value into a boolean isn't necessary.

Reply:
I was mainly trying to be consistent with the rest of boolean values in the client. It seems that all of them are parsed the same way. Check out `EnvVaultSkipVerify`, `EnvVaultSRVLookup`, and `EnvVaultInsecure `.

Reply:
This area of the code then is inconsistent with other parts of Vault 😞